transcendence-app/src/chat/chat.gateway.ts

```diff
@@ -30,10 +32,45 @@ export class ChatGateway {
   constructor(
     private chatService: ChatService,
     private chatroomMemberService: ChatroomMemberService,
+    private userService: UserService,
   ) {}
 
+  @SubscribeMessage('chatMessage')
+  async handleChatMessage(
+    @MessageBody()
+    createChatMessageDto: CreateChatMessageDto,
+    @ConnectedSocket() client: Socket,
+  ) {
+    const { userId: recipientId, content } = createChatMessageDto;
+
+    const senderId = client.request.user.id;
+
+    const block = await this.userService.getBlockRelation(
+      recipientId,
+      senderId,
+    );
+
+    if (block?.isUserBlockedByMe || block?.amIBlockedByUser) {
+      throw new WsException('There is an existing blockage. Forbidden.');
+    }
```

Comment:
question: We have an ongoing [task](https://github.com/42AGV/ft_transcendence/issues/330) about whether the blocked user should know who blocks them. What is your opinion on this matter?

Reply:
I won't waste time improving this. Although it would be really nice the subject only mentions about not being able to see blocked user messages and the simplest way is just to prevent accessing the chat conversation like is now

Reply:
Ok, then we could also close [#330](https://github.com/42AGV/ft_transcendence/issues/330)